src/Models/Sources.php

```diff
@@ -102,4 +103,42 @@ public function validateAppleUser(string $identityToken): object
 
         return $appleUserInfo;
     }
+
+    /**
+     * validation.
+     *
+     * @param  string $email
+     * @param  string $token
+     *
+     * @return bool
+     */
+    public function validation(string $email, string $token) : bool
+    {
+        $di = DI::getDefault();
+        try {
+            switch ($this->title) {
+                case 'google':
+                        $client = $di->getGoogle();
+                        $payload = $client->verifyIdToken($token);
+                        if ($payload) {
+                            $userid = $payload['sub'];
+                            return $payload['email'] === $email;
+                        } else {
+                            throw new Exception('Invalid user on google validation, payload or email incorrect');
+                        }
+                    break;
+                case 'facebook':
+                        $fb = $di->getFacebook();
+                        $response = $fb->get('/me', $token);
+                        $user = $response->getGraphUser();
+                        if ($user) {
+                            return true;
+                        }
+                        throw new Exception('Invalid user on facebook validation');
+                break;
+            }
+        } catch (Throwable $e) {
+            throw new Exception($e->getMessage());
```

Comment:
@FredPeal throw $e; instead of this

Reply:
@FredPeal why catch this execption here if you are just going to throw it again? let kanvas get it for you